frontend/src/views/AllDeclarationsPage/index.vue

```diff
@@ -28,28 +62,59 @@ import InstructionDeclarationsTable from "./InstructionDeclarationsTable"
 import { DsfrPagination } from "@gouvminint/vue-dsfr"
 import { useRoute, useRouter } from "vue-router"
 import { getPagesForPagination } from "@/utils/components"
+import { DsfrInput } from "@gouvminint/vue-dsfr"
 
 const router = useRouter()
 const route = useRoute()
 
 const hasDeclarations = computed(() => data.value?.count > 0)
 const showPagination = computed(() => data.value?.count > data.value?.results?.length)
-
-const updatePage = (newPage) => router.push({ query: { page: newPage + 1 } })
+const offset = computed(() => (page.value - 1) * limit)
 
 const limit = 10
-const page = computed(() => parseInt(route.query.page))
 const pages = computed(() => getPagesForPagination(data.value.count, limit, route.path))
+const openDeclaration = (id) => router.push({ name: "InstructionPage", params: { declarationId: id } })
+
+const statusFilterOptions = [
```

Comment:
Remarque pour plus tard : je suis un peu embêté comme toi à chaque fois avec les Enum Python qu'on doit avoir en doublon aussi côté front.
J'ai pas encore trouvé de bonne solution à ça. Peut être in fine un endpoint unique qui contient toutes les Enums du back et est appelé une unique fois au load de l'app, pour mettre ça dans le state ?